retriever.py

```diff
@@ -61,102 +121,80 @@ def __init__(self, experiment_name):
-        with torch.no_grad():
-            # full_model_states = self.model(
-            #     tokens, output_hidden_states=True
-            # )
-            # test_embedding = full_model_states.hidden_states[-1].detach()
-
-            _, func_embedding = self.model.forward(
-                tensor, 0, output_last_hidden_state=True, attn_mask=attn_mask
-            )
-            pooled_embedding = torch.sum(func_embedding, dim=1)
-
-            similarity_matrix = F.cosine_similarity(
-                self.embeddings, pooled_embedding
-            )
-            # Largest scores are last!
-            sorted_indices = torch.argsort(similarity_matrix, descending=False)
-
-            print("Top 10 Most Relevant Tests")
-            top_indices = sorted_indices[-10:]
-            for ind in top_indices:
-                test = self.unittest_names[int(ind.item())]
+            for ind in range(similarity_matrix.shape[0]):
+                test = self.unittest_names[ind]
                 score = similarity_matrix[ind]
-                print(f"Score: {score}; Test: {test}")
+                if test not in mapping:
+                    mapping[test] = []
+                mapping[test].append(score.item())
+
+        # condense
+        for test, score in mapping.items():
+            mapping[test] = sum(score) / len(score)
+        return mapping
 
-            print("Filtered Least Relevant Tests")
-            bot_indices = sorted_indices[:1000]
-            for ind in bot_indices:
-                test = self.unittest_names[int(ind.item())]
-                score = similarity_matrix[ind]
-                if "test" in test.split(":")[-1].lower():
-                    print(f"Score: {score}; Test: {test}")
+
+def gen_output_file(filename, mapping):
+    """Make json file of the mapping in assets/mappings"""
+    os.makedirs("assets/mappings", exist_ok=True)
+    new_mapping = {}
+    for file, score in mapping.items():
+        clean_file = os.path.relpath(file, REPO_ROOT.parent / "pytorch/test")
+        new_mapping[clean_file] = score
+    with open(REPO_ROOT / "assets/mappings" / filename, "w") as f:
+        f.write(json.dumps(new_mapping))
+        print(f"Made output file assets/mappings/{filename}")
 
 
 def main():
     parser = ArgumentParser("Retriever")
     parser.add_argument(
-        "--experiment-name",
-        type=str,
+        "--experiment-names",
+        nargs="+",
+        required=True,
+        help="Uses artifacts from the specified Indexer Experiments",
+    )
+    parser.add_argument(
+        "--pr-items",
+        nargs="+",
+        choices=PR_ITEMS.keys(),
         required=True,
-        help="Uses artifacts from the specified Indexer Experiment",
+        help="Specify what method to parse information from a PR",
     )
+
     args = parser.parse_args()
 
     start = time.time()
-    retriever = Retriever(args.experiment_name)
-    retriever.retrieve()
-    end = time.time()
+    experiments = {
```

Comment:
Do we need logic for different experiments in here? They should just be launched as different `torchrun` programs with different args to `pr-items`. Or we can have a separate wrapping script called experiments.py that calls these,  but I'd lean towards keeping the core indexer/retriever abstractions clean and minimal.